chips/nrf52/src/ieee802154_radio.rs

```diff
@@ -718,20 +753,15 @@ impl<'a> Radio<'a> {
     }
 
     fn rx(&self) {
-        self.registers.event_ready.write(Event::READY::CLEAR);
+        self.state.set(RadioState::RX);
 
-        if self.transmitting.get() {
-            let tbuf = self.tx_buf.take().unwrap(); // Unwrap fail = Radio TX Buffer produced an invalid result when setting the DMA pointer.
+        // Unwrap fail = Radio RX Buffer produced an invalid result when setting the DMA pointer.
+        let rbuf = self.rx_buf.take().unwrap();
+        self.rx_buf.replace(self.set_dma_ptr(rbuf));
 
-            self.tx_buf.replace(self.set_dma_ptr(tbuf));
-        } else {
-            let rbuf = self.rx_buf.take().unwrap(); // Unwrap fail = Radio RX Buffer produced an invalid result when setting the DMA pointer.
-            self.rx_buf.replace(self.set_dma_ptr(rbuf));
-        }
+        self.registers.shorts.write(Shortcut::READY_START::SET);
```

Comment:
You are correct in your assumption; the shortcuts are stateful. The reason for turning on/off the shortcuts when switching from receiving/transmitting/acknowledging is that they each require a different set of shortcuts that are not overlapping. I agree that a one time init would simplify the implementation. However, this would mean that we remove shortcuts and handle each progression of the state machine in the interrupt handler. This is entirely feasible, but will likely lead to a higher latency for each of our events. Does it make sense to have a higher latency for radio operations to simplify the state machine/readability of what the driver is doing?

Reply:
> that they each require a different set of shortcuts that are not overlapping.

🤯 😭 — okay, I didn't quite appreciate that the active shortcuts were changing on different paths. If that's the case, then as-is is likely best.